bindings/rust/extended/s2n-tls/src/enums.rs

```diff
@@ -103,6 +103,30 @@ impl TryFrom<s2n_tls_version::Type> for Version {
     }
 }
 
+#[non_exhaustive]
+#[derive(Debug, PartialEq, Copy, Clone)]
+pub enum CertSNIMatch {
+    /// The client did not supply an SNI
+    NoSNI,
+    ExactMatch,
+    WildcardMatch,
+    NoMatch,
+}
+
+impl TryFrom<s2n_cert_sni_match::Type> for CertSNIMatch {
+    type Error = Error;
+    fn try_from(input: s2n_cert_sni_match::Type) -> Result<Self, Self::Error> {
+        let version = match input {
```

Comment:
Nit: Eh I don't really think version is a good name for this variable.  Maybe just "match_type" is better?